### ISSUE TITLE
Handle module dependencies

### ISSUE THREAD

**Issue by nmanumr** [eraeco/pay#39]
Currently, our application loads multiple HTML modules asynchronously using jQuery's `.load()` method. However, some modules are dependent on others, which can cause issues when they're loaded out of order.

For example `accounts` module requires `remember` module to be loaded first because we need to inset `#pillbox` before `.remember`. 

https://github.com/eraeco/pay/blob/43879c32e32fb575bca1fabcae193f5b5de44a74/%2B/accounts.html#L76

We need to implement some solution that that either remove this kind of coupling between modules or/and ensures dependencies to be loaded before the dependant modules.

## Possible Solution

we can refactor our load function to return a promise which can be used as dependency of other modules. Here is how it may look like:

```js
async function load(url, deps = []) {
  await Promise.all(deps);
  const {promise, resolve, reject} = Promise.withResolvers();

  $("#tray").append(
    $("<div>").load(url, function (response, status, xhr) {
      if (status == "error") {
        reject(xhr.statusText);
      } else {
        $("#tray").append(div);
        resolve();
      }
    })
  );

  return promise
}

// then we can load it like this:
let rememberModule = load('./+/remember.html');
load('./+/help/deposit.html')
load('./+/metrics.html');
load('./+/marketing.html');
load('./+/paste.html');
load('./+/autoconvert.html');
load('./+/reflow.html');
load('./+/print.html');
// now accounts module will always load after remember module.
load('./+/accounts.html', [rememberModule]);
```

**Comment by nmanumr** [comment]
> True, but `reflow` doesn't have ordering issues and I noticed it would sometime work sometimes not - so there may still be another issue going on.

Ok let me investigate this bit more.

> A fix I want us to try first is to have parent modules load their sub-module: index loads `remember` which will load `accounts`. This'll help us create a tree going down. Rather than index loading both remember & accounts.

Yeah thats a nice idea and will work great for now. But the the issue with this approach will appear once we ha multiple modules dependent on once module. For example `ModuleB` and `ModuleC` depends on `ModuleA` now I we load `ModuleA` inside `ModuleB` and `ModuleC`. it will cause following issue:

1. `ModuleA` will be loaded twice (or atleast we need to dedup module loading)
2. It will create waterfall issue with modules loading (which i noticed in the proposed solution as well but can easily be fixed by only mounting the dependant module after dependencies are resolved), the network will look like this:

```
ModuleB: |-------->[Loaded]--------------------------->[waits for ModuleA and then mounts]
ModuleA:                   |---------->[Loaded]->[mounts]

ModuleC: |-------->[Loaded]--------------------------->[waits for ModuleA and then mounts]
ModuleA:                   |------->[Loaded]->[mounts]
```

instead of this
```
ModuleA: |---------->[Loaded]-->[mounts]
ModuleB: |------->[Loaded]-------------->[wait for Module A to be mounted and then mounts]
ModuleB: |----->[Loaded]---------------->[wait for Module A to be mounted and then mounts]
```


let me know what you think?

**Comment by nmanumr** [comment]
I'm able to reproduce the `reflow` issue as well this happens when [`load`](https://github.com/eraeco/pay/blob/43879c32e32fb575bca1fabcae193f5b5de44a74/index.html#L759) method is called after reflow module is loaded.

and a simple solution could be to emit a custom event when load is done and listen to that event from reflow module and retrigger `reflow`. This will keep reflow decoupled from `load` method.

Do you agree?

**Comment by amark** [comment]
whichever is easier... either (A) trigger onblur/keyup/onchange/oninput or (B) custom event. I lean more towards (A) since it seems like a more accurate simulation of what we want... or ?should? be less buggy since we do not have to REMEMBER to code for special custom events.

Module loading. I agree with what you're saying... but can we keep it submodule Tree style until we hit 2 submodules dependent on other/common module - then we'll deal with it then. Sorry to be boring.

**Comment by nmanumr** [comment]
Fixed by #40 and #41.
